### PR TITLE
String Manipulation and Comparison Operations

### PR DESCRIPTION
**Pull request by tonyta**
Hi again!

I thought I'd suggest some changes after reading through this beautiful source!

**Changes:**

1. Refactor some string manipulations so they are more performant (up to 3-4x faster) and more concise.
2. Swapped around some comparisons to be more conventional: with the (more) constant object on the right of the operator.

Hope this is helpful! 😸

### CODE REVIEW

File: lib/http/redirector.rb [hunk 1]
```diff
@@ -68,13 +68,13 @@ def perform(request, response)
     # Check if we reached max amount of redirect hops
     # @return [Boolean]
     def too_many_hops?
-      1 <= @max_hops && @max_hops < @visited.count
+      @max_hops > 0 && @visited.count > @max_hops
```
Comment (britishtea) [inline lib/http/redirector.rb]: I think `@visited.size` or `@visited.length` would be more descriptive here.

**Comment by ixti** [comment]
Agree with `#1` - I like what you've did to content type parsing and will cherry-pick it.
But I have pretty strong objections against second one. :D

File: lib/http/response/status.rb [hunk 2]
```diff
@@ -43,7 +43,7 @@ def coerce(object)
         # @param [#to_s] str
         # @return [Symbol]
         def symbolize(str)
-          str.to_s.downcase.tr("-", " ").gsub(/[^a-z ]/, "").gsub(/\s+/, "_").to_sym
+          str.to_s.downcase.tr("- ", "_").delete("^a-z_").to_sym
```
Comment (ixti) [inline lib/http/response/status.rb]:
```
[1] pry(main)> "Bad  request".downcase.tr("- ", "_").delete("^a-z_").to_sym
=> :bad__request
[2] pry(main)> "Bad  request".downcase.tr("-", " ").gsub(/[^a-z ]/, "").gsub(/\s+/, "_").to_sym
=> :bad_request
```

**Comment by ixti** [comment]
Partially merged in https://github.com/httprb/http/commit/015a7070a08ae41951d54bb143d7d0a5a2b8a794

**Comment by tonyta** [comment]
@ixti Thanks for the quick feedback. I'm just curious, what are your objections to swapping the variables?

**Comment by ixti** [comment]
Well, mostly pretty subjective personal vision of beauty :D I prefer to see constants on the left of expression. I guess this is actually the only reason :D after all...

**Pull request closed without merging**